src/utils/request/parseBody.ts

```diff
@@ -17,15 +17,15 @@ export function parseBody(body?: MockedRequest['body'], headers?: Headers) {
   // parse it into an object.
   const hasMultipartContent = contentType?.startsWith('multipart/form-data')
   if (hasMultipartContent && typeof body !== 'object') {
-    return parseMultipartData(body, headers) || body
+    return parseMultipartData(body.toString(), headers) || body
   }
 
   // If the intercepted request's body has a JSON Content-Type
   // parse it into an object.
   const hasJsonContent = contentType?.includes('json')
 
   if (hasJsonContent && typeof body !== 'object') {
-    return jsonParse(body) || body
+    return jsonParse(body.toString()) || body
```

Comment:
I wondered should I fix also `jsonParse` type to enable `jsonParse` to get also a number value,  because `JSON.parse` can handle a number value.

 Since `JSON.parse` converts its arguments to strings first in processing, I decided not to update `jsonParse`, feeling that this change is trivial in processing and not worth a complex change to the module interface.

FYI: https://tc39.es/ecma262/multipage/structured-data.html#sec-json.parse

FYI: https://tc39.es/ecma262/multipage/structured-data.html#sec-json.parse

Reply:
`JSON.parse` indeed accepts _anything_ and attempts to convert it to a string if it's not. I think this is a TS limitation of `JSON.parse` interface being typed as accepting strings only:

```ts
interface JSON {
    /**
     * Converts a JavaScript Object Notation (JSON) string into an object.
     * @param text A valid JSON string.
     * @param reviver A function that transforms the results. This function is called for each member of the object.
     * If a member contains nested objects, the nested objects are transformed before the parent object is.
     */
    parse(text: string, reviver?: (this: any, key: string, value: any) => any): any;
}
```

I think we should adjust the call signature of `jsonParse` to accept `any` instead of converting every input to a string (which is redundant and is only needed to please TS).

```diff
export function jsonParse<T extends Record<string, any>>(
-  str: string,
+  value: any,
): T | undefined {
```

This is a safe change, as `jsonParse` is added precisely to have a safe JSON parsing, which includes given a malformed input (expected scenario). Once we do this, we can get rid of `.toString()` calls anywhere we wish to parse a non-string data with `jsonParse`. _This_ can now be reflected in unit tests for `jsonParse`.